File: app/src/main/java/org/cru/godtools/adapter/BaseViewHolder.java
package org.cru.godtools.adapter;

import android.view.View;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import butterknife.ButterKnife;

abstract class BaseViewHolder extends RecyclerView.ViewHolder {
    BaseViewHolder(@NonNull final View rootView) {
        super(rootView);
        ButterKnife.bind(this, rootView);
    }

    void bind(final int position) {}
}

File: app/src/main/java/org/cru/godtools/adapter/BaseViewHolder.kt
package org.cru.godtools.adapter

import androidx.databinding.ViewDataBinding
import butterknife.ButterKnife
import org.ccci.gto.android.common.recyclerview.adapter.DataBindingViewHolder

internal abstract class BaseViewHolder<B : ViewDataBinding>(binding: B) : DataBindingViewHolder<B>(binding) {
    init {
        ButterKnife.bind(this, binding.root)
    }

    protected open fun bind(position: Int) = Unit
}
